### ISSUE TITLE
Signal crash on start

### ISSUE THREAD

**Issue by tubia** [signalapp/Signal-Android#4864]
Hi,
Two updates ago the application started to crash at the beginning at the password prompt.  
I'm not able to do anything as this happens immediately.
Is there any way I can see the logs somewhere?
I tried to set to airplane mode to check if it was related but no luck.

I could reinstall the app but I fear loss of data.

**Comment by agrajaghh** [comment]
`adb logcat` would be really helpfull

**Comment by tubia** [comment]
Yep.
[This](https://gist.github.com/tubia/47c3275cae9fcd1d5f77) is the output from the Signal App icon press to the crash and display of the error message.

**Comment by scienmind** [comment]
I have had about 3-4 very similar crushes over last week as well.

**Comment by moxie0** [comment]
@tubia did you update from a hella old version (like 20 releases?)

**Comment by tubia** [comment]
Hi @moxie0.  
No, I'm quite diligent in updating TextSecure/Signal (to limit problems communicating with friends) so I think the last update before this one was a couple of weeks ago.

**Comment by tubia** [comment]
So actually what kind of solution I could try?
I would like to save my messages if possible...
For what I can understand from the logs, it's some decrypt problem. However 10 days ago the application worked just fine.

**Comment by moxie0** [comment]
@tubla Before this started crashing, did you successfully have conversations with this contact in recent months?

**Comment by peterpanondrugs** [comment]
- 1st crash series when i tried to send someone (signal ios) and message which failed...
- then after 3-4 crashes signal was fine again. he showed me, that he failed to send a message.
- i wrote again that guy. again failed...but this time signal didnt start anymore. on clikcing the app symbol, signal would imm. crash.
- i deleted data! all "fine" again >_<

https://gist.github.com/anonymous/d41cd15b7c0f86d3732f

**Comment by tubia** [comment]
Hi @moxie0 when it happened, all conversation were working as they should. I updated recently but the problem persists. Well, definitely I suppose it'll work if I delete old data and reinstall but it's the think I really want to avoid.

**Comment by haffenloher** [comment]
@tubia I think he wanted to know whether you recently exchanged messages with that specific contact whose number is mentioned in [your log](https://gist.github.com/tubia/47c3275cae9fcd1d5f77) :) (after `java.lang.AssertionError: Session didn't get migrated`)

**Comment by tubia** [comment]
Hi @haffenloher thank you for the explanation!
I checked the number but I don't have it in my contacts. Maybe it is this the problem, that this contact has been added in a group but I don't have it in my contacts. Possible?

**Comment by tubia** [comment]
Well I tried to add this number to my contacts but it didn't work.
I found, however, when is triggered the error reported: https://github.com/WhisperSystems/Signal-Android/blob/master/src/org/thoughtcrime/securesms/crypto/storage/TextSecureSessionStore.java#L71
Maybe could be useful for the developers.

**Comment by aurelg** [comment]
Same problem here: Signal crashes upon start. ADB logcat reports a `java.lang.AssertionError: Session didn't get migrated`. It then refers to the phone number of a contact who I exchanged a few messages with, using Signal Desktop client. Since this recurrent crash, none of the messages I sent w/the Signal Desktop client, even to other people, were delivered. I removed the specific contact from the phone, the crash still happens. I put it back, and reset the session from the Signal Desktop client, no change. 

Any suggestion?
